### PR TITLE
feat(bank linking): refactor payment methods and saved accounts and a…

### PR DESCRIPTION
**Pull request by blockdylanb**
## Description (optional)
Refactors the payments methods UI in order to break out save cards/funds/bank accounts into a separate flyout than the payment method options.

## Testing Steps (optional)
Test buy flow with different payment options, saved, and new.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/selectors.ts [hunk 1]
```diff
@@ -154,6 +154,17 @@ export const getDefaultPaymentMethod = (state: RootState) => {
   return lift(transform)(sbCardsR, sbMethodsR, sbBalancesR)
 }
 
+export const hasFiatBalances = (state: RootState) => {
+  const fiatBalances = Object.keys(
+    state.components.simpleBuy.balances.data
+  ).filter(
+    currency =>
+      currency in FiatTypeEnum &&
```
Comment (dsisto-bc) [inline packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/selectors.ts:162]: Nice! Didn't know this syntax

File: packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/LinkedPaymentAccounts/Accounts/index.tsx [hunk 2]
```diff
@@ -182,8 +200,8 @@ class Payments extends PureComponent<InjectedFormProps<{}, Props> & Props> {
       value
     }))
 
-    const defaultCardMethod = this.props.paymentMethods.methods.find(
-      m => m.type === 'PAYMENT_CARD' && orderType === 'BUY'
+    const bankTransfer = defaultMethods.find(
+      method => method.value.type === 'BANK_TRANSFER' && orderType === 'BUY'
     )
```
Comment (dsisto-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/LinkedPaymentAccounts/Accounts/index.tsx:205]: It's before this change but I don't understand why `orderType === 'BUY'` is inside the `find`. It's either true or false for all items in the list.

File: packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/LinkedPaymentAccounts/Accounts/index.tsx [hunk 3]
```diff
@@ -218,10 +226,7 @@ class Payments extends PureComponent<InjectedFormProps<{}, Props> & Props> {
         card: card.card,
         type: 'USER_CARD',
         currency: card.currency,
-        limits:
-          defaultCardMethod && defaultCardMethod.limits
-            ? defaultCardMethod.limits
-            : { min: '1000', max: '500000' }
+        limits: { min: '1000', max: '500000' }
```
Comment (dsisto-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/LinkedPaymentAccounts/Accounts/index.tsx:229]: Don't know if you need them. But `BANK_TRANSFER` eligibility response has `limits`

File: packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/PaymentMethods/Methods/index.tsx [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+
+    const cardMethods = availableCards.map(card => ({
+      text: card.card
+        ? card.card.label
+          ? card.card.label
+          : card.card.type
+        : 'Credit or Debit Card',
+      value: {
+        ...card,
+        card: card.card,
```
Comment (dsisto-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/PaymentMethods/Methods/index.tsx:212]: I think `card: card.card` is included in `...card`

**Review by dsisto-bc — APPROVED** [review]
LGTM! I just pointed some stuff I didn't get. But nothing big

File: packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/PaymentMethods/Methods/index.tsx (hunk 4 above)
Comment (milan-bc) [reply packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/PaymentMethods/Methods/index.tsx:212]: Funny but not aways :( this is a definitely thing to be improved.

File: packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/EnterAmount/Checkout/Payment/index.tsx [hunk 5]
```diff
@@ -14,42 +14,51 @@ import {
...
-    </PaymentContainer>
-  </>
-)
+const Payment: React.FC<Props> = props => {
+  const step =
+    props.hasFiatBalance ||
+    props.cards.length ||
+    props.bankTransferAccounts.length
+      ? 'LINKED_PAYMENT_ACCOUNTS'
+      : 'PAYMENT_METHODS'
```
Comment (milan-bc) [inline packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/EnterAmount/Checkout/Payment/index.tsx:23]: cool 💯

**Review by milan-bc — APPROVED** [review]

File: packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/LinkedPaymentAccounts/Accounts/index.tsx (hunk 3 above)
Comment (schnogz) [reply packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/LinkedPaymentAccounts/Accounts/index.tsx:229]: yes, can we either fix this now or at least add a TODO comment? Need to do this soon

**Comment by milan-bc** [comment]
some conflicts to be fixed

**Comment by schnogz** [comment]
@blockdylanb can you address the conflicts and then merge into `feat/bank-linking`?

File: packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/selectors.ts (hunk 1 above)
Comment (blockdylanb) [reply packages/blockchain-wallet-v4-frontend/src/data/components/simpleBuy/selectors.ts:162]: Yeah, TBH I was hoping there was something in TS that was more like a set intersection so I wouldn't have to do the `filter` but I couldn't find anything. If you guys figure out a way please show me.

File: packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/LinkedPaymentAccounts/Accounts/index.tsx (hunk 3 above)
Comment (blockdylanb) [reply packages/blockchain-wallet-v4-frontend/src/modals/SimpleBuy/LinkedPaymentAccounts/Accounts/index.tsx:229]: I fixed this as part of another fix directly pushed the bank-linking branch. Might get a conflict with this code change.

**Pull request merged**